libs/domain/picking/src/components/navigate-back/navigate-back.component.ts

```diff
@@ -22,8 +22,14 @@ export class NavigateBackComponent
   @asyncState()
   protected link = valueType(this.routerService.previousRoute());
 
+  @asyncState()
+  protected currentLink = valueType(this.routerService.currentRoute());
+
   protected override render(): TemplateResult {
-    const link = this.link || this.fallbackUrl;
+    const link =
+      this.link && this.link !== this.currentLink
+        ? this.link
+        : this.fallbackUrl;
```

Comment:
Shouldn't we always navigate to the picking lists page instead? What's the case for navigating to a previous page?

Reply:
I believe this component was built with the thought of using it also in other places (in the future). That's why it doesn’t navigate always to `/`

Reply:
Oh true, but then it is accepting a prop for that so it can be customised easily where it's used.
So still my concern remains - can we remove all that complex router logic and just have a link prop?

Reply:
I guess we can do it and delegate the generating url logic to the parent component and provide it through the prop.